src/tasks/LibraryBuilder/Templates/autoinit.c

```diff
@@ -0,0 +1,162 @@
+#include <assert.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/system_properties.h>
+#include <unistd.h>
+
+#include <mono/jit/jit.h>
+#include <mono/jit/mono-private-unstable.h>
+#include <mono/metadata/assembly.h>
+#include <mono/metadata/mono-debug.h>
+
+static char *bundle_path;
+
+void register_aot_modules (void);
```

Comment:
Shouldn't we generate the registration code of aot modules direct in this file when building library instead of creating multiple C source files?

Reply:
The aot compiler generates the c file on our behalf. We could read that file in and place the contents here.

Reply:
Kept the aot module registration file (modules.c) generation in the MonoAOTCompiler as it gets placed in the Output Directory. The CMakeLists.txt of the LibraryBuilder is also generated in the same Output Directory, adds modules.c as an extra source (DOTNET_EXTRA_SOURCES), and links it  into the static lib `libaot_library.a`